superset/charts/dao.py

```diff
@@ -82,3 +84,34 @@ def favorited_ids(charts: List[Slice]) -> List[FavStar]:
             )
             .all()
         ]
+
+    @staticmethod
+    def user_slices(user_id: int) -> List[Any]:
+        owner_ids_query = (
+            db.session.query(Slice.id)
+            .join(Slice.owners)
+            .filter(security_manager.user_model.id == user_id)
+        )
+
+        return (
+            db.session.query(Slice, FavStar)
+            .join(
+                FavStar,
+                and_(
+                    FavStar.user_id == user_id,
+                    FavStar.class_name == "slice",
+                    Slice.id == FavStar.obj_id,
+                ),
+                isouter=True,
+            )
+            .filter(  # pylint: disable=comparison-with-callable
+                or_(
+                    Slice.id.in_(owner_ids_query),
```

Comment:
Any reason this isn’t a join as opposed to a subquery? I suspect the subquery likely fetches significantly more slices than the user favorited.